core/store/src/main/java/uk/gov/gchq/gaffer/store/schema/Schema.java

```diff
@@ -350,8 +348,10 @@ public CHILD_CLASS merge(final Schema schema) {
 
                     mergeConfig(thatSchema);
 
+                } catch (final SchemaException e) {
+                    throw e.preAppendToMessage(ERROR_MERGING_SCHEMA_DUE_TO);
```

Comment:
A little confused by this method when straight afterwards `throw new SchemaException(ERROR_MERGING_SCHEMA_DUE_TO + e.getMessage(), e);` is used

Reply:
This is used to add some additional information to any exceptions thrown that extend the `SchemaException` type without loosing the sub exception type which is caught and handled in the `MergeSchema` class.

Technically this is kind of the same as doing:

```java
throw e.getCause().getClass().getDeclaredConstructor().newInstance("something" + e.getMessage());`
```

Reply:
Okay, this does look less verbose than what you showed, so makes sense. Maybe naming could be better though

Reply:
I agree, to be honest it might be simpler now to just catch the exact exceptions now there is only three to eliminate the need for custom internal prepend method e.g.
```java
catch (SplitElementGroupDefException e) {
    throw new SplitElementGroupDefException("extra info" + e.getMessage(), e);
} catch (VertexSerialiserSchemaException e) {
    throw new VertexSerialiserSchemaException("extra info" + e.getMessage(), e);
} catch (VisibilityPropertySchemaException e) {
    throw new VisibilityPropertySchemaException("extra info" + e.getMessage(), e);
}
```

Reply:
But they add different information depending on the type

Reply:
This line is adding the same information.